File: exercises/101-109_small_problems/easy_01/easy_03.rb
# easy_03.rb

def digit_list(number)
   number.to_s.split(//).map { |num| num.to_i }
end

=begin

utp
- positive integer, and return an array

i/o
- input is an integer
- output will be an array whose elements are integers built from string

test cases

puts digit_list(12345) == [1, 2, 3, 4, 5]
puts digit_list(7) == [7]
puts digit_list(375290) == [3, 7, 5, 2, 9, 0]
puts digit_list(444) == [4, 4, 4]

algorithm
- convert integer to string, split string into an array of individual chars
- map array into integers and return
=end

File: exercises/101-109_small_problems/easy_01/java/Easy03.java
class Digits {
  int[] digitList(int num) {
    String[] strDigits = Integer.toString(num).split("");
    int[] digits = new int[strDigits.length];

    for (int i = 0; i < digits.length; i++) {
      digits[i] = Integer.parseInt(strDigits[i]);
    }

    return digits;
  }
}

class Easy03 {
  public static void main(String[] args) {
    Digits d = new Digits();
    int[] t1 = d.digitList(12345);
    int[] t2 = d.digitList(7);
    int[] t3 = d.digitList(375290);
    int[] t4 = d.digitList(444);

    for (int n : t1) {
      System.out.print(n + " ");
    }
    System.out.println();

    for (int n : t2) {
      System.out.print(n + " ");
    }
    System.out.println();

    for (int n : t3) {
      System.out.print(n + " ");
    }
    System.out.println();

    for (int n : t4) {
      System.out.print(n + " ");
    }
    System.out.println();
  }
}
